test/demo/eventListeners.inttest.ts

```diff
@@ -0,0 +1,170 @@
+import 'jest';
+import {Application, Window as FinWindow} from 'hadouken-js-adapter';
+
+import {NotificationClickedEvent, Notification, NotificationOptions, NotificationButtonClickedEvent, NotificationClosedEvent} from '../../src/client';
+
+import {fin} from './utils/fin';
+import * as notifsRemote from './utils/notificationsRemoteExecution';
+import {getCardsByNotification, isCenterShowing} from './utils/notificationCenterUtils';
+import {delay} from './utils/delay';
+
+const defaultNoteOptions: NotificationOptions = {
+    body: 'Test Notification Body',
+    title: 'Test Notification Title',
+    buttons: [
+        {title: 'Button 1'}
+    ]
+};
+
+describe('Click listeners', () => {
+    beforeAll(async () => {
+        // Make sure the center is showing
+        if (!(await isCenterShowing())) {
+            await notifsRemote.toggleNotificationCenter({uuid: 'test-app', name: 'test-app'});
+        }
+    });
+
+    describe('With one app running', () => {
+        let testApp: Application;
+        let testAppMainWindow: FinWindow;
+        beforeEach(async () => {
+            testApp = await createTestApp();
+            testAppMainWindow = await testApp.getWindow();
+            await delay(2000);
+        });
+
+        afterEach(async () => {
+            await testApp.close();
+            await delay(2000);
```

Comment:
This is quite a long teardown delay (and same for initialisation), does it need to be this long?

Do we want some `waitForX` utils? For example - wait for 100 ms in a loop, until `fin.System.getAllApplications` does/doesn't return a particular app identity? Should give a good compromise between correctness and execution time. Perhaps as part of the test util story (SERVICE-524)?

Reply:
All of the delays could do with a go-over to see what kinds of delays are actually needed. I think this one was just thrown in when I was debugging something else, so should be removable. Also, fwiw app.close() shouldn't, as far as I'm aware, resolve until the app is actually closed. The reason we have delays after it in other services it to let the provider clean up any state about the window which in this case is not needed.

Reply:
The only place something like that may be useful is when we call create and wait for the toast to appear, but that could probably be done as part of the toastUtils story

Reply:
After some discussion agreed that this would be covered under the toast testing story if it is deemed needed then.